packages/cdktf/lib/terraform-resource.ts

```diff
@@ -347,4 +381,84 @@ export class TerraformResource
   public addMoveTarget(moveTarget: string) {
     this._addResourceTarget(moveTarget);
   }
+
+  /**
+   * Moves this resource to the resource corresponding to "id"
+   * @param id Full id of resource to move to
+   */
+  public moveToId(id: string) {
+    if (this._movedById) {
+      throw new Error(`
+      ${this.node.id} has been given two separate moved operations.
+
+      {
+        from: ${this._movedById.from}
+        to: ${this._movedById.to}
+      }
+      {
+        from: ${id}
+        to: ${this.terraformResourceType}.${this.friendlyUniqueId} (Resource calling the move to operation)
+      }
+
+      Only one move operation can occur per plan/apply. Remove one of the operations.
+      `);
+    }
+    this._movedById = {
+      to: id,
+      from: `${this.terraformResourceType}.${this.friendlyUniqueId}`,
+    };
+    this._hasMoved = true;
+    this.node.addValidation(
+      new ValidateTerraformVersion(
+        ">=1.5",
+        `Resource move functionality is only supported for Terraform >=1.5. Please upgrade your Terraform version.`
+      )
+    );
+  }
+
+  /**
+   * Move the resource corresponding to "id" to this resource. Note that the resource being moved from must be marked as moved using it's instance function.
+   * @param id Full id of resource being moved from
+   */
+  public moveFromId(id: string) {
+    if (this._movedById) {
+      throw new Error(`
+      ${this.node.id} has been given two separate moved operations.
+
+      {
+        from: ${this._movedById.from}
+        to: ${this._movedById.to}
+      }
+      {
+        from: ${this.terraformResourceType}.${this.friendlyUniqueId} (Resource calling the move from operation)
+        to: ${id}
+      }
+
+      Only one move operation can occur plan/apply. Remove one of the operations.
+      `);
+    }
+    this._movedById = {
+      to: `${this.terraformResourceType}.${this.friendlyUniqueId}`,
+      from: id,
+    };
+    this.node.addValidation(
+      new ValidateTerraformVersion(
+        ">=1.5",
+        `Resource move functionality is only supported for Terraform >=1.5. Please upgrade your Terraform version.`
+      )
+    );
+  }
+
+  /**
+   * Marks this resource as moved. Required to be present on a resource being moved from.
+   */
```

Comment:
I see that the Aspect might require this, as it can't delete resources after they've been added to the construct tree.
If the user controls this, however, it would also be possible to remove the resource, right? We could hint at that other option in this doc string (if true)
```suggestion
  /**
   * Marks this resource as moved. Required to be present on a resource being moved from.
   * You can also remove the entire resource instead, if you don't need to programmatically decide whether to move it.
   * This gets set automatically if you use targets to move resources, no need to call this method in that case.
   */
```

Reply:
(just a suggestion, feel free to put your own spin to it)

Reply:
Aspect actually doesn't require it. 

The idea being that the user upgrade their CDKTF version, which in turn would give different resource addresses for everything. In the Aspect we use the old id generation logic to then move everything in place from those old ids that are now (in light of the upgrade) no longer present. Meaning `hasMoved` isn't actually needed in anyway.

Going to remove the convention as it's redundant.